### DISCUSSION TITLE
FR: main_guard improvement

### DISCUSSION THREAD

**Discussion by ershov** [HexFiend/HexFiend#466] [Ideas]
# Feature request

It would be really great if there was something like `main_guard` with better error reporting. Ideally, I'd like to see:

1. Byte offset of the error.
2. Nested `section` path of the error.
3. Stack trace.

**Comment by kainjow** [comment]
Sorry, I’m not fully understanding. Can you provide more detail?

**Comment by kainjow** [comment]
I now realize you're referring to the utility function in General.tcl. This was contributed by @fosterbrereton so I would suggest discussion with them.

**Comment by ershov** [comment]
I actually think that this can require support on the binary side.

The issue is that the error message only gives the error text like "error decoding blah". But imagine that you have a deep file structure and code with error recovery. Additional context would be incredibly helpful. It might include: (a) nested sections path; (b) TCL stack trace.
